tracee-rules/engine/engine.go

```diff
@@ -50,7 +56,9 @@ func NewEngine(sigs []types.Signature, sources EventSources, output chan types.F
 			if es.Source == "" {
 				engine.logger.Printf("signature %s doesn't declare an input source", meta.Name)
 			} else {
+				engine.signaturesMutex.Lock()
```

Comment:
why is `signatures` map not protected by the mutex?

Reply:
signatures and signaturesIndex are protected by the same lock as they refer to same objects and you want to be safe you don't have discrepancies between the data of the two data structures.

Reply:
I see that you've added the mutex around the L44. maybe just move it here under the same lock? not a big deal though